### ISSUE TITLE
Misleading error when network endpoints invalid

### ISSUE THREAD

**Issue by ClarenceL** [CityOfZion/neo-js#114]
I added an endpoint with the wrong port, got an odd error, when it's actually a wrong network endpoint passed in via the options

##### Error: index not defined
`node.js:72
            while ((this.blockWritePointer < this.mesh.highestNode().index) &&`

This originates in the `setInterval` loop in `node.js` - `deferredUpdateLoop`

Turns out Promise rejections from the `rpc` function `getBlockCount` are ignored, in `rpc.js` the `call` function returns a Promise.reject when there is an HTTP error, but there is no `promise.catch` in `getBlockCount`.

Then `mesh.highestNode()` returns `undefined` since that error caused the node's active flag to be set to false.

**Comment by rockacola** [comment]
Replicate concern. Error log:

```
/neo-js/dist/neo.js:138
          while ((this.blockWritePointer < this.mesh.getHighestNode().index) && (this.queue.length() < this.maxQueueLength)) {
                                                                     ^

TypeError: Cannot read property 'index' of undefined
    at Timeout.setInterval [as _onTimeout] (/neo-js/dist/neo.js:138:70)
    at ontimeout (timers.js:475:11)
    at tryOnTimeout (timers.js:310:5)
    at Timer.listOnTimeout (timers.js:270:5)
```

**Comment by rockacola** [comment]
Need a mechanism to check for at least 1 node to be valid/available before commence the syncing process.

**Comment by rockacola** [comment]
Prefer to explore this after we have event emitter (or observerable) in place.

**Comment by rockacola** [comment]
This has been properly addressed in coming `0.8.0` release. You should now see following error (when enabled) when there isn't any valid nodes available:

```
SyncStrategy error: Unable to find a valid node.
```

**Comment by rockacola** [comment]
Legacy issue from `<=0.8.1`.
